de.metas.adempiere.adempiere/base/src/main/java/de/metas/adempiere/service/impl/AddressBuilder.java

```diff
@@ -247,29 +249,37 @@ else if (outStr.lastIndexOf("\n") == outStr.length() - 1 && text.trim().length()
 			}
 			else if (token.equals("R"))
 			{
-				if (location.getC_Region() != null) // we have a region
+				if (location.getC_Region() != null)
+				{
 					outStr.append(location.getC_Region().getName());
+				}
 				else if (location.getRegionName() != null
 						&& location.getRegionName().length() > 0)
+				 {
 					outStr.append(location.getRegionName()); // local region name
+				}
 			}
 			else if (token.equals("P"))
 			{
 				if (location.getPostal() != null)
+				{
 					outStr.append(location.getPostal());
+				}
 			}
 			else if (token.equals("A"))
 			{
 				final String add = location.getPostal_Add();
 				if (add != null && add.length() > 0)
+				{
 					outStr.append("-").append(add);
+				}
 			}
 			else if (token.equals("CO"))
 			{
 				final String countryName;
 				if (isLocalAddress)
```

Comment:
why don't we translate the country name if `isLocalAddress==true`?

Reply:
@metas-ts Thinking now, I think is a bit more complicated.
Because we have client language and the system base language.
System base language is German, so actually in C_Country, the names are in German for all countries.
So you need to take the name from C_Country if local is de_DE, and from translation, if otherwise.
Correct?